maat-court-data-api/src/test/java/gov/uk/courtdata/dces/controller/FdcContributionsRestControllerTest.java

```diff
@@ -0,0 +1,79 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(FdcContributionsRestController.class)
+@AutoConfigureMockMvc(addFilters = false)
+class FdcContributionsRestControllerTest {
+
+    private static final String ENDPOINT_URL = "/api/internal/v1/debt-collection-enforcement";
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private FdcContributionsService fdcContributionsService;
+
+    @Test
+    void testContributionFileContent() throws Exception {
+        BigDecimal expectedCost1= new BigDecimal("100.1");
+        BigDecimal expectedCost2= new BigDecimal("444.44");
+        BigDecimal expectedCost3= new BigDecimal("999.99");
+
+        when(fdcContributionsService.getFdcContributionFiles(FdcContributionsStatus.REQUESTED))
+                .thenReturn(List.of(
+                        FdcContributionsResponse.builder().id(1).finalCost(expectedCost1).build(),
+                        FdcContributionsResponse.builder().id(2).finalCost(expectedCost2).build(),
+                        FdcContributionsResponse.builder().id(3).finalCost(expectedCost3).build()));
+
+        mvc.perform(MockMvcRequestBuilders.get(String.format(ENDPOINT_URL  +"/fdc-contribution-files"))
+                        .queryParam("status", FdcContributionsStatus.REQUESTED.name())
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(3)))
+                .andExpect(jsonPath("$.[?(@.id==1)].id").exists())
+                .andExpect(jsonPath("$.[?(@.id==1)].finalCost").value(expectedCost1.doubleValue()))
+                .andExpect(jsonPath("$.[?(@.id==2)].id").exists())
+                .andExpect(jsonPath("$.[?(@.id==2)].finalCost").value(expectedCost2.doubleValue()))
+                .andExpect(jsonPath("$.[?(@.id==3)].id").exists())
+                .andExpect(jsonPath("$.[?(@.id==3)].finalCost").value(expectedCost3.doubleValue()));
+    }
+
+    @Test
+    void testContributionFileContentWhenActiveFileNotAvailable() throws Exception {
+
+        when(fdcContributionsService.getFdcContributionFiles(FdcContributionsStatus.REQUESTED)).thenReturn(List.of());
+
+        mvc.perform(MockMvcRequestBuilders.get(String.format(ENDPOINT_URL  +"/fdc-contribution-files"))
+                        .queryParam("status", FdcContributionsStatus.REQUESTED.name())
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(0)));
+    }
+
+    @Test
+    void testContributionFileContentWhenQueryParamIsNotProvided() throws Exception {
+        mvc.perform(MockMvcRequestBuilders.get(String.format(ENDPOINT_URL  +"/fdc-contribution-files"))
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isBadRequest());
```

Comment:
Can you include an assertion on the body of the response also?

i.e.
`.andExpect(jsonPath("$.detail").value("Required parameter 'status' is not present."));`